### ISSUE TITLE
Local and remote migrations are not in lockstep

### ISSUE THREAD

**Issue by iherger** [prisma/migrate#46]
After an attempted migration, I get the following error:
`Error: Local and remote migrations are not in lockstep. We have migration 20190707191708 locally and 20190707213735 remotely at the same position in the history.`

How can this be fixed? Neither of `prisma migrate save/up/down` wants to do anything...

**Comment by pantharshit00** [comment]
Looks like a migration failed. Try deleting the entry from the _Migrations table which matches the error message.

If the error is persistent please give us a minimal reproduction

**Comment by artemzakharov** [comment]
I'm running into this issue too, nuking every row in the `_Migrations` table seems to allow me to apply my migration, but I still don't understand what the error means or how it's supposed to be solved correctly... what do you mean by "deleting the entry which matches the error message" @pantharshit00 ? The error message names two migrations, are you saying to delete both?

**Comment by pantharshit00** [comment]
Right now these need to be manually handled. Yes, I am saying to delete both of the entries.

In the near future, we will use the lift.lock file to resolve conflicts via the CLI.

**Comment by tvvignesh** [comment]
Facing the same issue on my end as well. This is the error I run the command in package.json script where my programs are running as pods in Kubernetes. So, 2 pods succeeded and 2 of them give this error.

```
> cd "src/app/db/prisma" && yes | prisma2 lift up

Error: Local and remote migrations are not in lockstep. We have migration 20191220182717-tcdb locally and 20191220124833-tcdb remotely at the same position in the history.
```

**Comment by pantharshit00** [comment]
@tvvignesh You should NOT apply migrations again and again. Please do it in a separate step in your CI pipeline.

**Comment by tvvignesh** [comment]
> @tvvignesh You should NOT apply migrations again and again. Please do it in a separate step in your CI pipeline.

Sure. Have separated it out now. Figuring out how to setup the CI for database without duplicating `schema.prisma` in both the service and the CI. Any specific recommendations available for CI pipeline setup for migrate? Any workflow people typically follow? In my case, I use Gitlab CI with Skaffold on Kubernetes for all my microservices and databases. Thanks.

**Comment by pantharshit00** [comment]
@tvvignesh I remember you opened a new issue regarding this. We will follow up there.

**Comment by Mohammed-Farid** [comment]
Any update about this issue?

**Comment by tomhoule** [comment]
Thanks for taking the time to report this issue!

We've recently released a Prisma Migrate Preview ([2.1.3.0 release notes](https://github.com/prisma/prisma/releases/tag/2.13.0)), which has quite a few changes compared to the previous experimental version. We believe this issue is no longer relevant in this new version, so we are closing this.

We would encourage you to try out Prisma Migrate Preview. If you come across this or any other issue in the preview version, please feel free to open a [new issue in prisma/prisma](https://github.com/prisma/prisma/issues/new/choose).
For general feedback on Prisma Migrate Preview, feel free to chime in [on this issue](https://github.com/prisma/prisma/issues/4531).
